### DISCUSSION TITLE
zod-to-json-schema

### DISCUSSION THREAD

**Discussion by StefanTerdell** [colinhacks/zod#563] [Show and tell]
Hey! Just saw the ecosystem paragraph in the readme. I've made a zod to jsonSchema converter [available here](https://github.com/StefanTerdell/zod-to-json-schema).

Thanks for making Zod! It's pretty awesome

**Comment by Brian-McBride** [comment]
Hey Stefan, 

I did something similar, with the aim to generate an OpenAPI JSON object (for a swagger interface)
It might be great to combine efforts. 

https://github.com/colinhacks/zod/discussions/535

I took a slightly different approach and used a map of functions for conversion. 
I also tried to create some typings that extend the schema. Like carrying over min, max for numbers. 

OpenAPI just extends JSON Schema, so there is a good chunk of crossover.

**Reply by StefanTerdell** [reply]
That's a very cool implementation! Inspired me to finally add string pattern support :)

Min/max etc. is a part of JsonSchema7 spec "natively".
Not sure what you mean by combining efforts?
What features does OpenAPI extend over JsonSchema7?

**Comment by scotttrinh** [comment]
@StefanTerdell Feel free to add your project in a PR, I'll help get it merged for you.

**Reply by StefanTerdell** [reply]
Cheers!

https://github.com/colinhacks/zod/pull/583

**Comment by sudhanshug16** [comment]
Loving both the packages @StefanTerdell!

I feel like it should be a part of Zod since they are interoperable (in my testing, might have some edge cases).

Would like this first class support: `toJsonSchema`, `fromJsonSchema`. I am sure I am not the only one. Happy to own/assist.

**Reply by stefan-wullems** [reply]
Hey, I would indeed love a fromJsonSchema :)
